drivers/adc/adc_stm32.c

```diff
@@ -1174,6 +1176,58 @@ static const struct adc_driver_api api_stm32_driver_api = {
 	.ref_internal = STM32_ADC_VREF_MV, /* VREF is usually connected to VDD */
 };
 
+/*
+ * Prefix default when not configured by the DTS
+ * depending on the serie (stm32F1 has none)
+ */
+#if defined(CONFIG_SOC_SERIES_STM32L1X) || \
+	defined(CONFIG_SOC_SERIES_STM32U5X)
+#define prescaler_default	LL_ADC_CLOCK_ASYNC_DIV
+#elif !defined(CONFIG_SOC_SERIES_STM32F1X)
+#define prescaler_default	LL_ADC_CLOCK_SYNC_PCLK_DIV
+#endif
+
+/* st_prescaler property requires 2 elements : clock ASYNC/SYNC and DIV */
+#define STM32_ADC_CLOCK(x, prop)	DT_INST_PROP_BY_IDX(x, prop, 0)
+#define STM32_ADC_DIV(x, prop)		DT_INST_PROP_BY_IDX(x, prop, 1)
+
+/*
+ * Macro to check the property is valid with two elements:
+ * 1st one is SYNC or ASYNC (not all series have both)
+ * 2nd one for the divider (not all series have all values)
+ */
+#define ADC_STM32_HAS_PROP(x, prop)					\
+	IF_ENABLED(DT_INST_NODE_HAS_PROP(x, prop),			\
+		(COND_CODE_1(						\
+			IS_EQ(DT_INST_PROP_LEN(x, prop), 2),		\
+			(BUILD_ASSERT((					\
+				(STM32_ADC_CLOCK(x, prop) == SYNC) ||	\
+				(STM32_ADC_CLOCK(x, prop) == ASYNC)),	\
+				"Invalid ADC "#prop" property")),	\
+			(BUILD_ASSERT((DT_INST_PROP_LEN(x, prop) == 2),	\
+				"Invalid ADC "#prop" property length")))))
```

Comment:
Ok, so following should work.
```
#define ADC_STM32_HAS_PROP(x, prop)					\
	IF_ENABLED(DT_INST_NODE_HAS_PROP(x, prop),			\
			(BUILD_ASSERT(					\
				(STM32_ADC_CLOCK(x, prop) == SYNC) ||	\
				(STM32_ADC_CLOCK(x, prop) == ASYNC),	\
				"Invalid ADC "#prop" property");	\
			BUILD_ASSERT((DT_INST_PROP_LEN(x, prop) == 2),	\
				"Invalid ADC "#prop" property length")));
```

Reply:
Also, macro is quite specific to st,adc-prescaler binding.
Naming should be adapted to make it more explicit:
`DT_STM32_CHECK_PRESC` or similar.

Reply:
changing to `ADC_STM32_CHECK_PRESC`

Reply:
change macro to remove the COND_CODE_1. 
Note that it must  first check the length then the 1st element (SYNC/ASYNC) else the 
`st,adc-prescaler = <3>;`  is not detected as a length error